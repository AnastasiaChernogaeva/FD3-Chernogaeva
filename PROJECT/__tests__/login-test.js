import React from 'react';
import renderer from 'react-test-renderer';
 
import Login from '../components/login';


test('работа Login', () => {
 
  // создаём тестовую версию компонента
  const component = renderer.create(
    <Login />
  );
 
  // получаем снэпшот (HTML-снимок) компонента для сверки, что вёрстка не испортилась
  let componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();
 
  // найдём в вёрстке компонента сами кнопки
 
 
  const buttonEleme1 = component.root.find( el =>(el.props.testPrL1=="buttOn")); 
//   const buttonEleme2 = component.root.find( el =>(el.props.testPrR2=="buttOn"));  

  const buttonElems=[buttonEleme1, /*buttonEleme2,*/];
  // находим все кнопки
 
  buttonElems.forEach(elem=>{elem.props.onClick(); // по очереди нажимаем на кнопки и делаем снимк(так два раза)
  
  // получаем уже изменённый снэпшот
  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();
  // и "нажмём" на неё
 
  elem.props.onClick();
  
  // получаем уже изменённый снэпшот
  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();
 
})
})  