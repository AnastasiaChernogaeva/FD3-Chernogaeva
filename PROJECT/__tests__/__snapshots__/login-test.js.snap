// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`работа Login 1`] = `
<div>
  <div
    className="Top_Buttons"
  >
    <form
      className="login"
      onChange={[Function]}
    >
      <h1>
        Вход в аккаунт
      </h1>
      <label
        htmlFor="MailId"
      >
        Электронная почта
      </label>
      <br />
      <br />
      <input
        id="MailId"
        onChange={[Function]}
        onKeyPress={[Function]}
        type="text"
        value=""
      />
      <span
        className="error"
      >
        
      </span>
      <br />
      <br />
      <label
        htmlFor="Password"
      >
        Пароль
      </label>
      <br />
      <br />
      <input
        id="Password"
        onChange={[Function]}
        onKeyPress={[Function]}
        type="password"
        value=""
      />
      <span
        className="error"
      >
        
      </span>
      <br />
      <br />
      <input
        disabled="disabled"
        onClick={[Function]}
        type="button"
        value="Войти"
      />
      <input
        onClick={[Function]}
        type="button"
        value="Забыли пароль"
      />
    </form>
  </div>
</div>
`;
